data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/reconciler/impl/ResourcesReconcilerImpl.java

```diff
@@ -50,137 +60,150 @@ private ResourcesReconcilerImpl(
     this.ingressReconcilerListener = ingressReconcilerListener;
     this.egressReconcilerListener = egressReconcilerListener;
 
-    this.cachedResources = new HashMap<>();
-    this.cachedEgresses = new HashMap<>();
+    this.cachedResources = new HashMap<>(CACHED_RESOURCES_INITIAL_CAPACITY);
+    this.cachedEgresses = new HashMap<>(egressReconcilerListener != null ? CACHED_RESOURCES_INITIAL_CAPACITY : 0);
   }
 
   @Override
-  public Future<Void> reconcile(
-    Collection<DataPlaneContract.Resource> newResources) {
+  public Future<Void> reconcile(final Collection<DataPlaneContract.Resource> newResources) {
+    if (isReconcilingIngress()) {
+      return reconcileIngress(newResources);
+    }
+    return reconcileEgress(newResources);
```

Comment:
So now we can't reconcile both at the same time anymore?

Reply:
Yes, we can't. Reconciling both it's not in our use case, so why keep supporting it? 
With those additions we need to much machinery to support it.

Reply:
ok I'm fine with that